cli/collection.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/config"
+	"github.com/sourcenetwork/defradb/datastore"
+)
+
+func MakeCollectionCommand(cfg *config.Config) *cobra.Command {
+	var txID uint64
+	var name string
+	var schemaID string
+	var versionID string
+	var cmd = &cobra.Command{
+		Use:   "collection [--name <name> --schema <schemaID> --version <versionID>]",
+		Short: "Interact with a collection.",
+		Long:  `Create, read, update, and delete documents within a collection.`,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
```

Comment:
Sweet :) I was hoping you'd be able to find a way to do it like this when I was commenting - forces consistency and we get all the features :)